cache/disk/disk.go

```diff
@@ -211,6 +209,60 @@ func New(dir string, maxSizeBytes int64, opts ...Option) (Cache, error) {
 // Non-test users must call this to expose metrics.
 func (c *diskCache) RegisterMetrics() {
 	c.lru.RegisterMetrics()
+
+	prometheus.MustRegister(c.gaugeCacheAge)
+
+	// Update the cache age metric on a static interval
+	// Note: this could be modeled as a GuageFunc that updates as needed
+	// but since the updater func must lock the cache mu, it was deemed
+	// necessary to have greater control of when to get the cache age
+	go c.pollCacheAge()
+}
+
+// Update metric every minute with the current age of the cache
+func (c *diskCache) pollCacheAge() {
+	ticker := time.NewTicker(60 * time.Second)
+	for ; true; <-ticker.C {
+		c.updateCacheAgeMetric()
+	}
+}
+
+// Get the back item in the cache, and store its atime in a metric
+func (c *diskCache) updateCacheAgeMetric() {
+	c.mu.Lock()
+
+	key, value := c.lru.PeakOldestCacheItem()
+	age := 0.0
+
+	if key != nil {
+		f := c.getElementPath(key, value)
+		ts, err := atime.Stat(f)
+
+		if err != nil {
+			log.Printf("ERROR: failed to determine cache age: %v, unable to stat %s", err, f)
```

Comment:
If this is the case, we'll set the metric to 0.0 below, which might trigger alarms. I guess we should set a bool here, and only update the metric below if the bool is true.

Reply:
Thats fine with me to only update it on success. I was thinking that if the cache was consistently unable to stat the file then we would not want to keep broadcasting a potentially very old previous value. I was thinking that most would use an average metric over time (30m?) for something like this and occasional and rare `0` values would be naturally filtered out.